test/e2e/runner/load.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Load generates transactions against the network until the given
+// context is cancelled.
+func Load(ctx context.Context, testnet *e2e.Testnet) error {
+	concurrency := 50
+	initialTimeout := 1 * time.Minute
+	stallTimeout := 15 * time.Second
+
+	chTx := make(chan types.Tx)
+	chSuccess := make(chan types.Tx)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Spawn job generator and processors.
+	logger.Info("Starting transaction load...")
+	started := time.Now()
+
+	go loadGenerate(ctx, chTx)
+
+	for w := 0; w < concurrency; w++ {
+		go loadProcess(ctx, testnet, chTx, chSuccess)
+	}
+
+	// Monitor successful transactions, and abort on stalls.
+	success := 0
+	timeout := initialTimeout
+	for {
+		select {
+		case <-chSuccess:
+			success++
+			timeout = stallTimeout
+		case <-time.After(timeout):
+			return fmt.Errorf("unable to submit transactions for %v", timeout)
+		case <-ctx.Done():
+			if success == 0 {
+				return errors.New("failed to submit any transactions")
+			}
+			logger.Info(fmt.Sprintf("Ending transaction load after %v txs (%.1f tx/s)...",
+				success, float64(success)/time.Since(started).Seconds()))
+			return nil
+		}
+	}
+}
+
+// loadGenerate generates jobs until the context is cancelled
+func loadGenerate(ctx context.Context, chTx chan<- types.Tx) {
+	for i := 0; i < math.MaxInt64; i++ {
+		// We keep generating the same 1000 keys over and over, with different values.
+		// This gives a reasonable load without putting too much data in the app.
+		id := i % 1000
+
+		bz := make([]byte, 2048) // 4kb hex-encoded
```

Comment:
Something to possibly add later would be dynamic/configurable tx sizes. 

I have talked to a user or two that did testing with increasing tx sizes every couple of minutes, and they started to see degradation and in some cases failures.

Reply:
Yeah, I'm not sure if stress testing is something we necessarily want to be part of this. I suppose we could use the same infrastructure for setting up the testnet, but for actually generating stress test loads we should have a separate tool.